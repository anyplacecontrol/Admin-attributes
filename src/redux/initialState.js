import {uiInitialState} from "./modules/uiRedux";
import {routingInitialState} from "./modules/routingRedux";
import {productsInitialState} from "./modules/productsRedux";
import {ordersInitialState} from "./modules/ordersRedux";
import {mobileOrdersInitialState} from "./modules/mobileOrdersRedux";
import {productViewInitialState} from "./modules/productViewRedux";
import {productItemViewInitialState} from "./modules/productItemViewRedux";
import {orderViewInitialState} from "./modules/orderViewRedux";
import {mobileOrderViewInitialState} from "./modules/mobileOrderViewRedux";
import {categoriesInitialState} from "./modules/categoriesRedux";
import {productItemsInitialState} from "./modules/productItemsRedux";
import {rolesInitialState} from "./modules/rolesRedux";
import {statusesInitialState} from "./modules/statusesRedux";
import {tagsInitialState} from "./modules/tagsRedux";
import {tagViewInitialState} from "./modules/tagViewRedux";
import {storesInitialState} from "./modules/storesRedux";
import {categoryViewInitialState} from "./modules/categoryViewRedux";
import {statusViewInitialState} from "./modules/statusViewRedux";
import {roleViewInitialState} from "./modules/roleViewRedux";
import {storeViewInitialState} from "./modules/storeViewRedux";
import {machinesInitialState} from "./modules/machinesRedux";
import {machineViewInitialState} from "./modules/machineViewRedux";
import {kiosksInitialState} from "./modules/kiosksRedux";
import {kioskViewInitialState} from "./modules/kioskViewRedux";
import {metricsInitialState} from "./modules/metricsRedux";
import {metricViewInitialState} from "./modules/metricViewRedux";
import {customersInitialState} from "./modules/customersRedux";
import {customerViewInitialState} from "./modules/customerViewRedux";
import {usersInitialState} from "./modules/usersRedux";
import {userViewInitialState} from "./modules/userViewRedux";
import {mobileUsersInitialState} from "./modules/mobileUsersRedux";
import {mobileUserViewInitialState} from "./modules/mobileUserViewRedux";
import {reportOrdersInitialState} from "./modules/reportOrdersRedux";
import {reportOrdersByStoreInitialState} from "./modules/reportOrdersByStoreRedux";
import {cookingTipsInitialState} from "./modules/cookingTipsRedux";
import {cookingTipsViewInitialState} from "./modules/cookingTipsViewRedux";
import {alarmsInitialState} from "./modules/alarmsRedux";
import {alarmsViewInitialState} from "./modules/alarmsViewRedux";
import {reportInventoryByProductInitialState} from "./modules/reportInventoryByProductRedux";
import {reportInventoryByStoreInitialState} from "./modules/reportInventoryByStoreRedux";
import {reportProductsByPopularityInitialState} from "./modules/reportProductsByPopularityRedux";
import {reportTemperatureHistoryInitialState} from "./modules/reportTemperatureHistoryRedux";
import {reportTemperaturesByStoreInitialState} from "./modules/reportTemperaturesByStoreRedux";
import {reportSensorsHistoryInitialState} from "./modules/reportSensorsHistoryRedux";
import {dashboardInitialState} from "./modules/dashboardRedux";
import {controlsInitialState} from "./modules/controlsRedux";
import {controlViewInitialState} from "./modules/controlViewRedux";

export const initialState = {
  ui: uiInitialState,
  products: productsInitialState,
  productItems: productItemsInitialState,
  orders: ordersInitialState,
  mobileOrders: mobileOrdersInitialState,
  routing: routingInitialState,
  productView: productViewInitialState,
  productItemView: productItemViewInitialState,
  categories: categoriesInitialState,
  categoryView: categoryViewInitialState,
  orderView: orderViewInitialState,
  mobileOrderView: mobileOrderViewInitialState,

  statusesProducts: statusesInitialState,
  statusesProductItems: statusesInitialState,
  statusesOrders: statusesInitialState,
  statusesStores: statusesInitialState,
  statusesMachines: statusesInitialState,
  statusesKiosks: statusesInitialState,
  statusesCustomers: statusesInitialState,
  statusView: statusViewInitialState, //common for all statuses

  rolesAdmins: rolesInitialState,
  roleView: roleViewInitialState, //common for all statuses

  tags: tagsInitialState,
  tagView: tagViewInitialState,
  stores: storesInitialState,
  storeView: storeViewInitialState,
  machines: machinesInitialState,
  machineView: machineViewInitialState,
  kiosks: kiosksInitialState,
  kioskView: kioskViewInitialState,
  metrics: metricsInitialState,
  metricView: metricViewInitialState,
  controls: controlsInitialState,
  controlView: controlViewInitialState,
  users: usersInitialState,
  userView: userViewInitialState,
  customers: customersInitialState,
  customerView: customerViewInitialState,
  mobileUsers: mobileUsersInitialState,
  mobileUserView: mobileUserViewInitialState,

  reportOrders: reportOrdersInitialState,
  reportOrdersByStore: reportOrdersByStoreInitialState,
  cookingTips: cookingTipsInitialState,
  cookingTipsView: cookingTipsViewInitialState,
  alarms: alarmsInitialState,
  alarmsView: alarmsViewInitialState,
  reportInventoryByProduct: reportInventoryByProductInitialState,
  reportInventoryByStore: reportInventoryByStoreInitialState,
  reportProductsByPopularity: reportProductsByPopularityInitialState,
  reportTemperatureHistory: reportTemperatureHistoryInitialState,
  reportSensorsHistory: reportSensorsHistoryInitialState,
  reportTemperaturesByStore: reportTemperaturesByStoreInitialState,

  dashboard: dashboardInitialState,  
};
