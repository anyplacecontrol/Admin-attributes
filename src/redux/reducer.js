import {combineReducers} from "redux";
import ui from "./modules/uiRedux";
import products from "./modules/productsRedux";
import routing from "./modules/routingRedux";
import orders from "./modules/ordersRedux";
import mobileOrders from "./modules/mobileOrdersRedux";
import productView from "./modules/productViewRedux";
import productItemView from "./modules/productItemViewRedux";
import categories from "./modules/categoriesRedux";
import orderView from "./modules/orderViewRedux";
import mobileOrderView from "./modules/mobileOrderViewRedux";
import productItems from "./modules/productItemsRedux";
import statuses from "./modules/statusesRedux";
import roles from "./modules/rolesRedux";
import {
  statusesProducts,
  statusesProductItems,
  statusesOrders,
  statusesStores,
  statusesMachines,
  statusesKiosks,
  statusesCustomers
} from "./modules/statusesRedux";
import {
  rolesAdmins,
} from "./modules/rolesRedux";
import tags from "./modules/tagsRedux";
import tagView from "./modules/tagViewRedux";
import stores from "./modules/storesRedux";
import machines from "./modules/machinesRedux";
import kiosks from "./modules/kiosksRedux";
import categoryView from "./modules/categoryViewRedux";
import statusView from "./modules/statusViewRedux";
import roleView from "./modules/roleViewRedux";
import storeView from "./modules/storeViewRedux";
import machineView from "./modules/machineViewRedux";
import kioskView from "./modules/kioskViewRedux";
import metrics from "./modules/metricsRedux";
import metricView from "./modules/metricViewRedux";
import users from "./modules/usersRedux";
import userView from "./modules/userViewRedux";
import customers from "./modules/customersRedux";
import customerView from "./modules/customerViewRedux";
import mobileUsers from "./modules/mobileUsersRedux";
import mobileUserView from "./modules/mobileUserViewRedux";
import reportOrders from "./modules/reportOrdersRedux";
import reportOrdersByStore from  "./modules/reportOrdersByStoreRedux";
import cookingTips from "./modules/cookingTipsRedux";
import cookingTipsView from "./modules/cookingTipsViewRedux";
import alarms from "./modules/alarmsRedux";
import alarmsView from "./modules/alarmsViewRedux";
import logs from "./modules/logsRedux";
import logsView from "./modules/logsViewRedux";
import reportInventoryByProduct from "./modules/reportInventoryByProductRedux";
import reportInventoryByStore from "./modules/reportInventoryByStoreRedux";
import reportProductsByPopularity from "./modules/reportProductsByPopularityRedux";
import reportTemperatureHistory from "./modules/reportTemperatureHistoryRedux";
import reportSensorsHistory from "./modules/reportSensorsHistoryRedux";
import reportTemperaturesByStore from "./modules/reportTemperaturesByStoreRedux";
import dashboard from "./modules/dashboardRedux";
import controls from "./modules/controlsRedux";
import controlView from "./modules/controlViewRedux";

const reducers = {
  ui,
  products,
  productItems,
  routing,
  orders,
  mobileOrders,
  productView,
  productItemView,
  categories,
  categoryView,
  orderView,
  mobileOrderView,
  rolesAdmins: roles(rolesAdmins),
  statusesProducts: statuses(statusesProducts),
  statusesProductItems: statuses(statusesProductItems),
  statusesOrders: statuses(statusesOrders),
  statusesStores: statuses(statusesStores),
  statusesMachines: statuses(statusesMachines),
  statusesKiosks: statuses(statusesKiosks),
  statusesCustomers: statuses(statusesCustomers),
  statusView,
  roleView,
  storeView,
  tags,
  tagView,
  stores,
  machines,
  machineView,
  kiosks,
  kioskView,
  metrics,
  metricView,
  controls,
  controlView,
  users,
  userView,
  customers,
  customerView,
  mobileUsers,
  mobileUserView,
  cookingTips,
  cookingTipsView,
  alarms,
  alarmsView,
  logs,
  logsView,

  reportOrders,
  reportOrdersByStore,
  reportInventoryByProduct,
  reportInventoryByStore,
  reportProductsByPopularity,
  reportTemperatureHistory,
  reportSensorsHistory,
  reportTemperaturesByStore,

  dashboard
};

export default combineReducers(reducers);
