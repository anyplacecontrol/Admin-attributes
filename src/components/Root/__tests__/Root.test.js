import React from "react";
import Root from "../Root";
import renderer from "react-test-renderer";
import { configureStoreProd, history } from "../../../redux/configureStore";

//*******************************************************************************************************
describe(">>>Root --- mock store and mock App", () => {
    
  let store = configureStoreProd({});

  let wrapper = renderer.create(<Root history={history} store={store}/>);

  it("+++ render the component App: sleep page. No any modals", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

//*******************************************************************************************************
