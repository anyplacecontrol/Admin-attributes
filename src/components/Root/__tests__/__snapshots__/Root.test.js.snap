// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Root --- mock store and mock App +++ render the component App: sleep page. No any modals 1`] = `
<div
  className="sleep"
>
  <div
    className="sleep__top flex w100r animated"
    onClick={[Function]}
  >
    <span
      className="wrapper ph-30 animated"
    >
      <img
        className="sleep__image animated"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <span
        className="sleep__touch flex animated"
      >
        <img
          className="sleep__icon animated"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </span>
    </span>
  </div>
  <div
    className="sleep__bottom dark-blue w100r animated"
  >
    <div
      className="sleep__bottom--inner flex wrapper ph-30 animated"
    >
      <div
        className="sleep__title animated"
        onClick={[Function]}
      >
        Touch to Start
      </div>
      <button
        className="sleep__info handicapped flex animated"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="icon handicapped"
          className="sleep__info--icon animated"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </button>
    </div>
  </div>
</div>
`;
